source/client/hud.qc

```diff
@@ -1934,6 +1934,56 @@ void(float width, float height) HUD_RoundStopWatch =
 	Draw_String([width - (getTextWidth(stopwatch, 12)) - 2, 16], stopwatch, [12, 12], [1, 1, 1], 1, 0);
 }
 
+/*******************
+*  HUD Debug Info  *
+*******************/
+
+// set from CSQC_Ent_Update
+vector player_velocity;
+
+// probably only need these right here
+static void(vector pos, string label, vector v) Draw_Vector_Fancy =
+{
+	Draw_String(pos, label, [12, 12], [1, 1, 0], 1, 0);
+	Draw_String(pos + [32, 16], sprintf("X: %d", v.x), [12, 12], [1, 1, 1], 1, 0);
+	Draw_String(pos + [32, 32], sprintf("Y: %d", v.y), [12, 12], [1, 1, 1], 1, 0);
+	Draw_String(pos + [32, 48], sprintf("Z: %d", v.z), [12, 12], [1, 1, 1], 1, 0);
+};
+
+static void(vector pos, string label, float f, string suffix) Draw_Float_Fancy =
+{
+	Draw_String(pos, label, [12, 12], [1, 1, 0], 1, 0);
+	Draw_String(pos + [32, 16], sprintf("%d %s", f, suffix), [12, 12], [1, 1, 1], 1, 0);
+};
+
+void(float width, float height) HUD_PlayerDebugInfo =
+{
+	static float lastupstime;
+	static float lastups;
+
+	if (!cvar("scr_playerdebuginfo"))
+		return;
+
+	if ((time - lastupstime) >= 1.0 / 20)
```

Comment:
Is this `20` the tickrate? Not sure what this represents

Reply:
i copied that setup directly from FTE's source: https://github.com/fte-team/fteqw/blob/bf6ddf21ae1785ea5bb160cc5a5a6b4c3afbcd6d/engine/client/sbar.c#L2989-L3021 

there it seems rather arbitrary. i can change it if needed.

Reply:
Works for me! Thanks